dotCMS/src/main/java/com/dotcms/webdav/FsDirectoryResource.java

```diff
@@ -0,0 +1,263 @@
+        FileOutputStream out = null;
+        try {
+            out = new FileOutputStream(dest);
+            IOUtils.copy(in, out);
+        } finally {
+            IOUtils.closeQuietly(out);
+        }
+        // todo: ignores contentType
+        return factory.resolveFile(this.host, dest);
+
+    }
+
+    @Override
+    protected void doCopy(File dest) {
+        try {
+            FileUtils.copyDirectory(this.getFile(), dest);
+        } catch (IOException ex) {
+            throw new RuntimeException("Failed to copy to:" + dest.getAbsolutePath(), ex);
+        }
+    }
+
+    @Override
+    public LockToken createAndLock(String name, LockTimeout timeout, LockInfo lockInfo) throws NotAuthorizedException {
+        File dest = new File(this.getFile(), name);
+        createEmptyFile(dest);
+        FsFileResource newRes = new FsFileResource(host, factory, dest, contentService);
+        LockResult res = newRes.lock(timeout, lockInfo);
+        return res.getLockToken();
+    }
+
+    private void createEmptyFile(File file) {
+        FileOutputStream fout = null;
+        try {
+            fout = new FileOutputStream(file);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        } finally {
+            IOUtils.closeQuietly(fout);
+        }
+    }
+
+    /**
+     * Will generate a listing of the contents of this directory, unless the
+     * factory's allowDirectoryBrowsing has been set to false.
+     *
+     * If so it will just output a message saying that access has been disabled.
+     *
+     * @param out
+     * @param range
+     * @param params
+     * @param contentType
+     * @throws IOException
+     * @throws NotAuthorizedException
+     */
+    @Override
+    public void sendContent(OutputStream out, Range range, Map<String, String> params, String contentType) throws IOException, NotAuthorizedException {
+        String subpath = getFile().getCanonicalPath().substring(factory.getRoot().getCanonicalPath().length()).replace('\\', '/');
+        String uri = subpath;
+        //String uri = "/" + factory.getContextPath() + subpath;
+        XmlWriter w = new XmlWriter(out);
+        w.open("html");
+        w.open("head");
+        w.writeText(""
+                + "<script type=\"text/javascript\" language=\"javascript1.1\">\n"
+                + "    var fNewDoc = false;\n"
+                + "  </script>\n"
+                + "  <script LANGUAGE=\"VBSCRIPT\">\n"
+                + "    On Error Resume Next\n"
+                + "    Set EditDocumentButton = CreateObject(\"SharePoint.OpenDocuments.3\")\n"
+                + "    fNewDoc = IsObject(EditDocumentButton)\n"
+                + "  </script>\n"
+                + "  <script type=\"text/javascript\" language=\"javascript1.1\">\n"
+                + "    var L_EditDocumentError_Text = \"The edit feature requires a SharePoint-compatible application and Microsoft Internet Explorer 4.0 or greater.\";\n"

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Split this 171 characters long line (which is greater than 150 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00103)